topics/icrc_X_multiple_canister_call.md

```diff
@@ -0,0 +1,237 @@
+            
+        else Rejected
+            U ->> S: Reject request
+            S ->> U: Display reject message
+            S ->> RP: Error response: Action aborted (3001)
+        end
+    end
+```
+
+### Examples
+
+#### Successful Batch Call
+
+Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrcX_batch_call_canisters",
+  "params": {
+    "mode": "sequence",
+    "sender": "b7gqo-ulk5n-2kpo7-oalt7-p2kyl-o4j5l-kiuwo-eeybr-dab4l-ur6up-pqe",
+    "requests": [
+      {
+        "id": 1,
+        "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+        "method": "transfer",
+        "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+      },
+      {
+        "id": 2,
+        "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+        "method": "transfer",
+        "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+      }
+    ]
+  }
+}
+```
+
+Response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "responses": [
+      {
+        "id": 1,
+        "result": {
+          "Ok": "16509194"
+        }
+      },
+      {
+        "id": 2,
+        "result": {
+          "Ok": "16509195"
+        }
+      }
+    ]
+  }
+}
+```
+
+#### Batch Call with Error
+
+Request
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "method": "icrcX_batch_call_canisters",
+  "params": {
+    "mode": "",
+    "sender": "b7gqo-ulk5n-2kpo7-oalt7-p2kyl-o4j5l-kiuwo-eeybr-dab4l-ur6up-pqe",
+    "isAnonymous": false,
+    "requests": [
+      {
+        "id": 1,
+        "params": {
+          "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+          "method": "transfer",
+          "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+        }
+      },
+      {
+        "id": 2,
+        "params": {
+          "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+          "method": "transfer",
+          "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+        }
+      },
+      ,
+      {
+        "id": 3,
+        "params": {
+          "canisterId": "xhy27-fqaaa-aaaao-a2hlq-ca",
+          "method": "transfer",
+          "arg": "RElETARte24AbAKzsNrDA2ithsqDBQFsA/vKAQKi3pTrBgHYo4yoDX0BAwEdV+ztKgq7E4l1ffuTuwEmw8AtYSjlrJ+WLO5ofQIAAMgB"
+        }
+      }
+    ]
+  }
+}
+```
+
+Response
+
+```json
+{
+  "id": 1,
+  "jsonrpc": "2.0",
+  "result": {
+    "responses": [
+      {
+        "id": 1,
+        "result": {
+          "version": "1",
+          "Ok": "16509194"
+        }
+      },
+      {
+        "id": 2,
+        "result": {
+          "Err": {
+            "InsufficientFunds": {
+              "balance": {
+                "e8s": "0"
+              }
+            }
+          }
+        }
+      },
+      {
+        "id": 3,
+        "error": {
+          "code": 1000,
```

Comment:
Would be nice to also have an example where the processing of calls has stopped in between after an error with a call:

- result -> ...
- result -> ...
- error -> code: 4000
- error -> code: 1001
- error -> code: 1001